unlock-app/src/components/interface/checkout/main/Confirm/ConfirmCrossmint.tsx

```diff
@@ -0,0 +1,246 @@
+import {
+  CrossmintPaymentElement,
+  useCrossmintEvents,
+} from '@crossmint/client-sdk-react-ui'
+import { CheckoutService } from './../checkoutMachine'
+import { Connected } from '../../Connected'
+import { useConfig } from '~/utils/withConfig'
+import { Fragment, useState } from 'react'
+import { useActor } from '@xstate/react'
+import { PoweredByUnlock } from '../../PoweredByUnlock'
+import { Pricing } from '../../Lock'
+import { getReferrer, lockTickerSymbol } from '~/utils/checkoutLockUtils'
+import { Lock } from '~/unlockTypes'
+import { RiErrorWarningFill as ErrorIcon } from 'react-icons/ri'
+import { ViewContract } from '../../ViewContract'
+// import { useUpdateUsersMetadata } from '~/hooks/useUserMetadata'
+import { usePricing } from '~/hooks/usePricing'
+import { usePurchaseData } from '~/hooks/usePurchaseData'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { ethers } from 'ethers'
+import { useCrossmintEnabled } from '~/hooks/useCrossmintEnabled'
+import { TransactionAnimation } from '../../Shell'
+
+interface Props {
+  injectedProvider: unknown
+  checkoutService: CheckoutService
+  onConfirmed: (lock: string, hash?: string) => void
+  onError: (message: string) => void
+}
+
+interface CrossmintQuote {
+  lineItems: any[]
+  totalPrice: {
+    amount: string
+    currency: string
+  }
+}
+
+export function ConfirmCrossmint({
+  injectedProvider,
+  onConfirmed,
+  // onError,
+  checkoutService,
+}: Props) {
+  const { email } = useAuth()
+  const [state] = useActor(checkoutService)
+  const config = useConfig()
+  const [isConfirming, setIsConfirming] = useState(false)
+  const [quote, setQuote] = useState<CrossmintQuote | null>(null)
+  const crossmintEnv = config.env === 'prod' ? 'production' : 'staging'
+
+  const {
+    lock,
+    recipients,
+    paywallConfig,
+    // metadata, // Should we save it?
```

Comment:
I don't think there is any need. We save it previously.

Reply:
agreed. It was there in the `ConfirmCrypto` component though. I will clean it up.

Reply:
Here is the cleanup PR: https://github.com/unlock-protocol/unlock/pull/12658/files